rmw_connextdds_common/src/common/rmw_impl.cpp

```diff
@@ -2982,6 +3032,21 @@ RMW_Connext_PublisherStatusCondition::get_status(
 {
   rmw_ret_t rc = RMW_RET_ERROR;
 
+  // If condition is attached to a waitset, lock its mutex to prevent
```

Comment:
same as above

Reply:
Removed in [b35d211](https://github.com/ros2/rmw_connextdds/pull/22/commits/b35d2114ef3b072f7ba386d1406eff89756d2288)